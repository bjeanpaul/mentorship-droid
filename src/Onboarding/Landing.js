import React, {
  Component,
} from 'react';

import {
  View,
  Text,
} from 'react-native';


import { styles } from '../StyleSheet';
import { Button } from '../Components/Button';

const Landing = () => (
  <View style={{ backgroundColor: '#dfe5e6', flex: 1 }}>
    <View style={{ flex: 1 }}>
    </View>
    <Text style={[styles.heading]}>
      This is a short, but sweet, description of the application.
    </Text>
    <Button labelText="Get Started" style={{ marginTop: 20, marginBottom: 20 }} />

    <View style={{ height: 48, backgroundColor: 'rgba(42, 45, 48, 0.05)', justifyContent: 'center' }}>
      <Text style={{ color: '#9fb1b3', fontSize: 14, textAlign: 'center' }}>
        Already have an account? <Text style={{ color: 'rgba(246, 111, 63, 0.9)' }}>Log in.</Text>
      </Text>
    </View>
  </View>
);

export default Landing;
