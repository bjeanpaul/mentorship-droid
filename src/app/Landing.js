import React from 'react';
import { View, Image, StyleSheet } from 'react-native';
import { BaseView, Text, Link, Button } from 'src/components';


const IMG_LANDING = require('app/assets/Landing.png');

const styles = StyleSheet.create({
  image: {
    flex: 0.6,
    overflow: 'hidden',
    alignItems: 'center',
  },
  getStarted: {
    padding: 15,
    backgroundColor: 'rgba(255, 255, 255, 0.5)',
  },
  getStartedHeading: {
    fontSize: 20,
    marginBottom: 15,
  },
  login: {
    padding: 15,
    backgroundColor: 'rgba(42, 45, 48, 0.05)',
    justifyContent: 'center',
  },
});

const Landing = () => (
  <BaseView>
    <View style={styles.image}>
      <Image resizeMode="stretch" source={IMG_LANDING} />
    </View>
    <View style={styles.getStarted}>
      <Text style={styles.getStartedHeading}>
        This is a short, but sweet, description of the application.
      </Text>
      <Button
        label="Get Started"
        handlePress={() => {
          // TODO: Transition
        }}
      />
    </View>
    <View style={styles.login}>
      <Text style={styles.loginText}>
        Already have an account?
        <Link> Log in.</Link>
      </Text>
    </View>
  </BaseView>
);

export default Landing;
