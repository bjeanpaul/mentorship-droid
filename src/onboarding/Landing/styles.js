import { StyleSheet } from 'react-native';
import colors from 'src/constants/colors';

export default StyleSheet.create({
  image: {
    flex: 0.6,
    overflow: 'hidden',
    alignItems: 'center',
  },
  getStarted: {
    flex: .5,
    padding: 15,
    backgroundColor: colors.ONBOARDING_LANDING_GET_STARTED_TEXT,
  },
  getStartedHeading: {
    fontSize: 20,
    marginBottom: 15,
  },
  login: {
    padding: 15,
    backgroundColor: colors.ONBOARDING_LANDING_LOGIN_BG,
    justifyContent: 'center',
  },
});
