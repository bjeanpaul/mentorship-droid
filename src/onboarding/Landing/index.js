import React, { PropTypes } from 'react';
import { View, Image } from 'react-native';
import { BaseView, Text, Link, Button, Icon } from 'src/components';

import styles from './styles';
import images from 'src/constants/images';

const Landing = ({
  onGetStartedPress,
  onLoginPress,
}) => (
  <BaseView>
    <View style={styles.imageContainer}>
      <Image
        style={styles.image}
        source={images.ONBOARDING_LANDING}
      />
    </View>
    <View style={styles.bottom}>
      <View style={styles.getStarted}>
        <Text style={styles.getStartedHeading}>
          This is a short, but sweet, description of the application.
        </Text>
        <Button
          theme={Button.themes.transparent}
          onPress={onGetStartedPress}
        >
          GET STARTED
        </Button>
      </View>
      <View style={styles.login}>
        <Text style={styles.loginText}>
          Already have an account?
          <Link onPress={onLoginPress}> Log in.</Link>
        </Text>
      </View>
    </View>
  </BaseView>
);
Landing.propTypes = {
  onGetStartedPress: PropTypes.func.isRequired,
  onLoginPress: PropTypes.func.isRequired,
};

export default Landing;
