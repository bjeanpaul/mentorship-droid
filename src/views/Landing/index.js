import React, { PropTypes } from 'react';
import { View, Image } from 'react-native';
import { BaseView, Text, Link, Button } from 'src/components';

import { MultiLineTextInput } from 'src/components';

import styles from './styles';
import images from 'src/constants/images';

const Landing = ({
  onGetStartedPress,
  onLoginPress,
}) => (
  <BaseView>

    <MultiLineTextInput
      placeholder="Type your answer here"
      maxLength={50}
    />


  </BaseView>
);
Landing.propTypes = {
  onGetStartedPress: PropTypes.func.isRequired,
  onLoginPress: PropTypes.func.isRequired,
};

export default Landing;
