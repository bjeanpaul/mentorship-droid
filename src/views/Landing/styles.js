import { merge } from 'lodash';
import { StyleSheet } from 'react-native';
import { DEVICE_HEIGHT, DEVICE_HEIGHT_SMALL } from 'src/constants/styles';
import colors from 'src/constants/colors';


export default StyleSheet.create(merge({
  imageContainer: {
    flex: 1,
    overflow: 'hidden',
    flexDirection: 'column',
    justifyContent: 'flex-start',
  },
  image: {
    flex: 1,
    width: null,
    height: null,
    resizeMode: 'cover',
  },
  bottom: {
    justifyContent: 'flex-end',
  },
  getStarted: {
    paddingHorizontal: 7,
    paddingTop: 7,
    paddingBottom: 7,
    backgroundColor: colors.ONBOARDING_LANDING_GET_STARTED_TEXT,
  },
  getStartedHeading: {
    paddingBottom: 11,
    fontSize: 20,
  },
  login: {
    paddingTop: 7,
    paddingBottom: 8,
    backgroundColor: colors.ONBOARDING_LANDING_LOGIN_BG,
    justifyContent: 'center',
    alignItems: 'center',
  },
}, DEVICE_HEIGHT > DEVICE_HEIGHT_SMALL && {
  getStarted: {
    paddingHorizontal: 24,
    paddingTop: 20,
    paddingBottom: 17,
  },
  getStartedHeading: {
    paddingBottom: 17,
    fontSize: 16,
  },
  login: {
    paddingTop: 15,
    paddingBottom: 17,
  },
}));
