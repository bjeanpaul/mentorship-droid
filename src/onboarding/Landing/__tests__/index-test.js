import React from 'react';
import { shallow } from 'enzyme';
import Landing from '../index';

describe('Landing', () => {
  function foo() {}
  function bar() {}

  it('should map `onGetStartedPress` and `onLoginPress`', () => {
    expect(
      <Landing
        onGetStartedPress={foo}
        onLoginPress={bar}
      />
    ).toMatchSnapshot();
  });

  it('should be able to tap and fire `onGetStartedPress`', () => {
    const onGetStartedPress = jest.fn();
    const el = shallow(
      <Landing
        onGetStartedPress={onGetStartedPress}
        onLoginPress={bar}
      />
    );
    el.find('Button').simulate('press');
    expect(onGetStartedPress).toBeCalled();
  });

  it('should be able to tap and fire `onLoginPress`', () => {
    const onLoginPress = jest.fn();
    const el = shallow(
      <Landing
        onGetStartedPress={foo}
        onLoginPress={onLoginPress}
      />
    );
    el.find('Link').simulate('press');
    expect(onLoginPress).toBeCalled();
  });
});
