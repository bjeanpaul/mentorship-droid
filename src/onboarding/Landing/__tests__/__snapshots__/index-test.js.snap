exports[`Landing should map \`onGetStartedPress\` and \`onLoginPress\` 0`] = `
Object {
  "$$typeof": Symbol(react.element),
  "_owner": null,
  "_store": Object {},
  "key": null,
  "props": Object {
    "onGetStartedPress": [Function foo],
    "onLoginPress": [Function bar]
  },
  "ref": null,
  "type": [Function Landing]
}
`;
